zwave_js_server/client.py

```diff
@@ -156,15 +179,18 @@ async def connect(self) -> None:
                 # Still adding it here to make sure we can always reconnect
                 self._logger.exception("Unexpected error")
 
-            if self.state == STATE_CONNECTED and self._on_disconnect:
-                await gather_callbacks(
-                    self._logger, "on_disconnect", self._on_disconnect
-                )
+            if self.state == STATE_CONNECTED:
+                # change state to connecting
+                self.state = STATE_CONNECTING
```

Comment:
I think we need to check `self.close_requested` here too and if that's true the state should be `STATE_DISCONNECTED`. That will handle the client getting disconnected cleanly by the user. In that case we don't want to say that we're connecting to callbacks.